OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -296,6 +296,14 @@ public WDist GetMaximumRangeVersusTarget(Target target)
 		// Enumerates all armaments, that this actor possesses, that can be used against Target t
 		public IEnumerable<Armament> ChooseArmamentsForTarget(Target t, bool forceAttack)
 		{
+			// If the target is frozen, make sure it is still valid
+			if (t.Type == TargetType.FrozenActor)
+			{
+				var frozenLayer = self.Owner.PlayerActor.TraitOrDefault<FrozenActorLayer>();
+				if (frozenLayer == null || frozenLayer.FromID(t.FrozenActor.ID) == null)
```

Comment:
I am querying it to make sure that it is still alive. When the units explored the fog and removed the frozen actor, it's object still exists, but is no longer in the layer, this state is what I'm trying to get with the "if".